[receiver/dockerstats] Remove featuregate and old scraper implementation

jamesmoessis opened the pull request:
**Description:** 
* Removes featuregate  that was marked stable last release. 
* Removes old implementation that featuregate toggled
* Removes deprecated config option `provide_per_core_cpu_metrics`
* There's no new code. Any green lines are things that I shuffled around so everything is in `receiver.go` (removed `metrics.go` and `receiver_v2.go`).

**Link to tracking Issue:** Closes https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/9794

**Testing:** 
No behaviour changes. Same metrics are still produced, and this is validated by existing tests. 

**Documentation:** 
Updated README.

jamesmoessis committed:
remove featuregate and old scraper implementation

jamesmoessis committed:
add changelog

MovieStoreGuy commented on receiver/dockerstatsreceiver/receiver.go:
```diff
@@ -88,29 +92,210 @@ func (r *receiver) scrape(ctx context.Context) (pmetric.Metrics, error) {
+	}
+}
+
+// From container.calculateCPUPercentUnix()
+// https://github.com/docker/cli/blob/dbd96badb6959c2b7070664aecbcf0f7c299c538/cli/command/container/stats_helpers.go
+// Copyright 2012-2017 Docker, Inc.
+// This product includes software developed at Docker, Inc. (https://www.docker.com).
+// The following is courtesy of our legal counsel:
+// Use and transfer of Docker may be subject to certain restrictions by the
+// United States and other governments.
+// It is your responsibility to ensure that your use and/or transfer does not
+// violate applicable laws.
+// For more information, please see https://www.bis.doc.gov
+// See also https://www.apache.org/dev/crypto.html and/or seek legal counsel.
+func calculateCPUPercent(previous *dtypes.CPUStats, v *dtypes.CPUStats) float64 {
+	var (
+		cpuPercent = 0.0
+		// calculate the change for the cpu usage of the container in between readings
+		cpuDelta = float64(v.CPUUsage.TotalUsage) - float64(previous.CPUUsage.TotalUsage)
+		// calculate the change for the entire system between readings
+		systemDelta = float64(v.SystemUsage) - float64(previous.SystemUsage)
+		onlineCPUs  = float64(v.OnlineCPUs)
+	)
+
+	if onlineCPUs == 0.0 {
+		onlineCPUs = float64(len(v.CPUUsage.PercpuUsage))
+	}
+	if systemDelta > 0.0 && cpuDelta > 0.0 {
+		cpuPercent = (cpuDelta / systemDelta) * onlineCPUs * 100.0
+	}
+	return cpuPercent
+}
+
+func calculateMemoryPercent(memoryStats *dtypes.MemoryStats) float64 {
+	if float64(memoryStats.Limit) == 0 {
+		return 0
+	}
+	return 100.0 * (float64(memoryStats.Usage) - float64(memoryStats.Stats["cache"])) / float64(memoryStats.Limit)
 }
```
Would it be worth moving to its own file named `thirdparty.go` just so it is clear on this is the only code copied and modified?

jamesmoessis replied:
This was just moved from `metrics.go` in this PR. Happy to move it elsewhere if required.

MovieStoreGuy reviewed (approved):
Changes look fine to me :) 

Just an outstanding question that @jpkrohling might be able to answer?

jamesmoessis committed:
remove old config from integration test

jamesmoessis committed:
go mod tidy

mx-psi reviewed (changes requested):
Putting this on hold until we resolve the conversation over at #17981 (specifically see https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/17981#discussion_r1084727981 ), we may want to keep the feature flag definition in the code (even if we remove the old implementation)

mapno committed:
[processor/servicegraph] Fix cache cleanup to also purge stale series (#17512)

Fix cache cleanup in servicegraph proccesor to also purge stale series

dgoscn committed:
[receiver/purefb] New Pure Storage FlashBlade receiver (#17906)

MovieStoreGuy committed:
[chore] Fixing up references (#18018)

Fix makefile reference

dashpole committed:
Remove jsumners from codeowners (#18034)

eedorenko committed:
[exporter/azuremonitor] Add metrics support (#14916)

frzifus committed:
[processor/resourcedetectionprocessor] support tls on openshift (#17963)

* [internal/metadataproviders] support tls settings to fetch openshift data

Signed-off-by: Benedikt Bongartz <bongartz@klimlive.de>

* [processor/resourcedetectionprocessor] support tls config

Signed-off-by: Benedikt Bongartz <bongartz@klimlive.de>

* chloggen: add changelog

Signed-off-by: Benedikt Bongartz <bongartz@klimlive.de>

Signed-off-by: Benedikt Bongartz <bongartz@klimlive.de>

frzifus committed:
codeowners: add frzifus to jaeger components (#17520)

Signed-off-by: Benedikt Bongartz <bongartz@klimlive.de>

Signed-off-by: Benedikt Bongartz <bongartz@klimlive.de>

eedorenko committed:
[exporter/azuremonitor] Fix CI (#18038)

See #14916 (comment)

my-git9 committed:
[extension] Cleanup ioutil for new go version (#18026)

Signed-off-by: xin.li <xin.li@daocloud.io>

nslaughter committed:
[sshcheckreceiver] New receiver (#14694)

This provides a remote check for the availability of SSH and SFTP servers. Observability for the status of SSH and SFTP servers is an important capability for practically all operators managing data for business customers.

gjanco committed:
[exporter/mezmoexporter] Remove specific path requirement (#18012)

* [exporter/mezmoexporter] Remove specific path requirement

* [exporter/mezmoexporter] Remove specific path requirement

Fixing unit tests

evan-bradley committed:
[chore] Add triaging processes and definitions to CONTRIBUTING.md (#18036)

Co-authored-by: Antoine Toulme <antoine@toulme.name>
Co-authored-by: Alex Boten <alex@boten.ca>

Commit:
[chore] update dependabot pr script (#18035)

This removes the need for the "Skip changelog" label as the title already contains "[chore]".

Signed-off-by: Alex Boten <aboten@lightstep.com>

bogdandrutu committed:
Change otelcontribcol and oteltestbedcol to use the builder (#16828)

Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

atoulme committed:
[exporter/splunkhecexporter] expose HTTPClientSettings on splunkhecexporter (#16839)

jefchien committed:
[exporter/awsxray] Add support for AWS X-Ray annotations attribute in exporter. (#17855)

Add support for AWS X-Ray annotations attribute in AWS X-Ray exporter.

Add to generated spans in AWS X-Ray receiver.

dmitryax committed:
[pkg/pdatatest] Bring reported errors to a consistent state (#18041)

This change applies several changes on reported errors:
- Instead of mixing different level errors (resource/scope/etc) and the same level errors with multierr, use prefixes to represent levels.
- Return as many errors as possible
- Bring errors messaging to a consistent concise view 
- Use lower case for field names to avoid linter complains

atoulme committed:
[receiver/haproxy] promote haproxy receiver to alpha (#18023)

atoulme committed:
[receiver/haproxy] add haproxy to components (#18021)

atoulme committed:
[chore] refactor the signalfxexporter internals to shutdown on component shutdown (#17987)

albertteoh committed:
[processor/spanmetrics] Fix flaky test (#18024)

* Fix flaky test from observed logs assertion

Signed-off-by: albertteoh <see.kwang.teoh@gmail.com>

dmitryax committed:
[chore] [pkg/pdatatest] Update instrumentation scope references (#18048)

fatsheep9146 committed:
[pkg/pdatatest] add more test cases for trace (#18031)

Signed-off-by: Ziqi Zhao <zhaoziqi9146@gmail.com>

liustanley committed:
[exporter/datadog] Support ddtags when batching logs (#17399)

* Change SubmitLogs to submit a new request for each log in the payload

* Optimize code for consecutive tag batching

* Changelog

* Add comment

* Refactor code

* Add comments and change variable names

* Clean up code

* Remove else block

* Change test names

* Fix import

* Move comment

my-git9 committed:
[all] Enable unused linter on golangci-lint (#18027)

Signed-off-by: xin.li <xin.li@daocloud.io>

kovrus committed:
[headerssetter] Extend  headerssetter with header actions. (#16818)

dmitryax committed:
[pkg/pdatatest] Ensure all fields are checked (#18050)

dmitryax committed:
[chore] [CI] Fix automation making a release (#18043)

djaglowski committed:
[receiver/snmp] Add start timestamp to sum metrics (#18002)

* [receiver/snmp] Add start timestamp to sum metrics

Commit:
[chore] add prepare-release action/script (#18057)

This will reduce the amount of time spent releasing for maintainers.

Signed-off-by: Alex Boten <aboten@lightstep.com>

djaglowski committed:
[connector/count] Add outline and documentation (#18037)

* [connector/count] Add outline and documentation

hanjm committed:
exporter/clickhouse: use endpoint instead of raw dsn in configuration. (#18067)

pmcollins committed:
[receiver/file] add file receiver skeleton 1/3 (#16827)

* add file receiver skeleton

bryan-aguilar committed:
[processor/groupbytrace] Add num_workers doc to README (#17305)

* Add num_workers doc to README

* Update gbt README with defaults and num_workers descript

* Update config.go to remove physical processor recomendation

* Add GOMAXPROCS starting point

frzifus committed:
[processor/resourcedetectionprocessor] fix readme layout and extend openshift info (#18071)

Signed-off-by: Benedikt Bongartz <bongartz@klimlive.de>

Signed-off-by: Benedikt Bongartz <bongartz@klimlive.de>

mx-psi committed:
[*/datadog] Bump Datadog Agent dependencies (#18074)

* [*/datadog] Bump Datadog Agent dependencies

* Address breaking change in public API of pkg/trace

evan-bradley committed:
[chore] Add additional debug logging to code owner workflows (#17569)

Adds additional debug logging to workflows that ping code owners without always requiring human interaction. This should help provide information to debug these workflows when something goes wrong, and can help with local testing.

Co-authored-by: Evan Bradley <evan-bradley@users.noreply.github.com>

liustanley committed:
[chore] Remove cmd/tracegen (#18077)

Co-authored-by: Dmitrii Anoshin <anoshindx@gmail.com>

frzifus committed:
[extension/jaegerremotesampling] remove misleading agent endpoint (#18070)

* [extension/jaegerremotesampling] remove misleading agent endpoint

Signed-off-by: Benedikt Bongartz <bongartz@klimlive.de>

* Update .chloggen/remove_misleading_jaegerremotesampling_endpoint.yaml

---------

Signed-off-by: Benedikt Bongartz <bongartz@klimlive.de>

MovieStoreGuy committed:
[chore] Adding new line escape (#18092)

Adding new line escape

gbbr committed:
exporter/datadogexporter: rename exportTraceMetrics to exportHostMetrics (#18094)

We don't want for it to be confused with APM Stats. This method is
unrelated to APM Stats, which are on occasion referred to as "trace
metrics".

rmfitzpatrick committed:
http_forwarder: specify beta stability and add codeowner (#18100)

songy23 committed:
[exporter/datadog] Fix a nil dereferencing in http response (#18102)

* [exporter/datadog] Fix a nil dereferencing in http response

* Add to changelog

wiardvanrij committed:
[exporter/loki] Do not retry on 4xx status code (excluding 429) (#18083)

opentelemetrybot committed:
[chore] dependabot updates Mon Jan 30 18:39:49 UTC 2023 (#18181)

Bump github.com/aws/aws-sdk-go from 1.44.186 to 1.44.189 in /internal/aws/cwlogs
Bump github.com/aws/aws-sdk-go from 1.44.186 to 1.44.189 in /internal/metadataproviders
Bump github.com/aws/aws-sdk-go-v2/config from 1.18.8 to 1.18.10 in /exporter/awskinesisexporter
Bump github.com/aws/aws-sdk-go-v2/config from 1.18.8 to 1.18.10 in /extension/sigv4authextension
Bump github.com/aws/aws-sdk-go-v2/credentials from 1.13.8 to 1.13.10 in /exporter/awskinesisexporter
Bump github.com/aws/aws-sdk-go-v2/credentials from 1.13.8 to 1.13.10 in /extension/sigv4authextension
Bump github.com/aws/aws-sdk-go-v2/service/sts from 1.18.0 to 1.18.2 in /extension/sigv4authextension
Bump github.com/sijms/go-ora/v2 from 2.5.24 to 2.5.25 in /receiver/sqlqueryreceiver
Bump github.com/snowflakedb/gosnowflake from 1.6.16 to 1.6.17 in /receiver/sqlqueryreceiver
Bump github.com/tencentcloud/tencentcloud-sdk-go/tencentcloud/common from 1.0.582 to 1.0.583 in /exporter/tencentcloudlogserviceexporter
Bump github.com/vmware/govmomi from 0.30.0 to 0.30.1 in /receiver/vcenterreceiver
Bump go.opentelemetry.io/otel from 1.11.2 to 1.12.0 in /receiver/opencensusreceiver
Bump go.opentelemetry.io/otel from 1.11.2 to 1.12.0 in /receiver/solacereceiver
Bump go.opentelemetry.io/otel/exporters/otlp/otlptrace from 1.11.2 to 1.12.0 in /cmd/telemetrygen
Bump go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracegrpc from 1.11.2 to 1.12.0 in /cmd/telemetrygen
Bump go.opentelemetry.io/otel/sdk from 1.11.2 to 1.12.0 in /cmd/telemetrygen
Bump go.opentelemetry.io/otel/sdk from 1.11.2 to 1.12.0 in /examples/demo/client
Bump go.opentelemetry.io/otel/trace from 1.11.2 to 1.12.0 in /cmd/telemetrygen
Bump go.opentelemetry.io/otel/trace from 1.11.2 to 1.12.0 in /pkg/ottl
Bump google.golang.org/grpc from 1.52.1 to 1.52.3 in /cmd/telemetrygen
Bump google.golang.org/grpc from 1.52.1 to 1.52.3 in /exporter/coralogixexporter
Bump google.golang.org/grpc from 1.52.1 to 1.52.3 in /exporter/skywalkingexporter
Bump google.golang.org/grpc from 1.52.1 to 1.52.3 in /extension/asapauthextension
Bump google.golang.org/grpc from 1.52.1 to 1.52.3 in /extension/basicauthextension
Bump google.golang.org/grpc from 1.52.1 to 1.52.3 in /extension/headerssetterextension
Bump google.golang.org/grpc from 1.52.1 to 1.52.3 in /extension/jaegerremotesampling
Bump google.golang.org/grpc from 1.52.1 to 1.52.3 in /extension/oauth2clientauthextension
Bump google.golang.org/grpc from 1.52.1 to 1.52.3 in /extension/sigv4authextension
Bump google.golang.org/grpc from 1.52.1 to 1.52.3 in /receiver/googlecloudspannerreceiver
Bump google.golang.org/grpc from 1.52.1 to 1.52.3 in /receiver/skywalkingreceiver

* make genotelcontribcol (#18183)
* make genoteltestbedcol (#18184)
---------

Co-authored-by: Alex Boten <aboten@lightstep.com>

atoulme committed:
[exporter/splunkhec] export log body as raw (#18082)

Support exporting log body to raw Splunk HEC endpoint

aboguszewski-sumo committed:
feat: remove elasticsearch feature gates for indices and cluster shards (#18189)

boostchicken committed:
[receiver/datadog] Add Datadog receiver (#5836)

Signed-off-by: John <john.dorman@sony.com>
Signed-off-by: John <dorman@overlooked.us>
Signed-off-by: John Dorman <john@boostchicken.dev>
Co-authored-by: Pablo Baeyens <pbaeyens31+github@gmail.com>
Co-authored-by: Gabriel Aszalos <gabriel.aszalos@gmail.com>
Co-authored-by: Juraci Paixão Kröhling <juraci.github@kroehling.de>
Co-authored-by: Sean Marciniak <30928402+MovieStoreGuy@users.noreply.github.com>
Co-authored-by: Alex Boten <alex@boten.ca>

boostchicken committed:
[receiver/datadog] Add Datadog receiver (#5836)

Signed-off-by: John <john.dorman@sony.com>
Signed-off-by: John <dorman@overlooked.us>
Signed-off-by: John Dorman <john@boostchicken.dev>
Co-authored-by: Pablo Baeyens <pbaeyens31+github@gmail.com>
Co-authored-by: Gabriel Aszalos <gabriel.aszalos@gmail.com>
Co-authored-by: Juraci Paixão Kröhling <juraci.github@kroehling.de>
Co-authored-by: Sean Marciniak <30928402+MovieStoreGuy@users.noreply.github.com>
Co-authored-by: Alex Boten <alex@boten.ca>

mar4uk committed:
Add promtail receiver (#18072)

* [pkg/translator/loki] use loki/pkg/push dependency, remove logproto copy

* [receiver/promtail] bring back promtailreceiver

loomis-relativity committed:
[receiver/azureeventhub] default to azure formatter (#18196)

* default to azure formatter

bogdandrutu committed:
[signalfxexporter] move config validation to Validate (#18205)

Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

Commit:
[influxdb] update InfluxDB receiver and exporter (#18080)

Update InfluxDB dependencies (influxdbreceiver and influxdbexporter). Also update influxexporter to adopt to breaking API changes in the github.com/influxdata/influxdb-observability line protocol converter and writer.

bogdandrutu committed:
[signalfxexporter] move initialization of the config defaults in Unmarshal. (#18207)

Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

songy23 committed:
[exporter/datadog] Bump datadog-agent dependencies to v0.43.0-rc.3 (#18206)

[exporter/datadog] bump datadog-agent dependencies to v0.43.0-rc.3

fatsheep9146 committed:
[pkg/pdatatest] replace compare of span event and link from reflect.DeepEqual to compare functions (#18088)

Signed-off-by: Ziqi Zhao <zhaoziqi9146@gmail.com>

atoulme committed:
[receiver/azureeventhub] add to components (#18209)

bogdandrutu committed:
[k8sclusterreceiver] Change internal metric store to use pdata (#18214)

This PR allows to change independently each metrics getter to pdata.

Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

mx-psi committed:
[*/datadog] Bump Datadog Agent modules (#18231)

* [*/datadog] Bump Datadog Agent modules

* Also processor/datadog

fatsheep9146 committed:
[pkg/pdatatest] add fatsheep9146 as owner (#18217)

Signed-off-by: Ziqi Zhao <zhaoziqi9146@gmail.com>

damemi committed:
[exporter/googlecloud] Update GCP dependencies (v0.35.0/v1.11.0) (#18182)

* Bump gcp dependencies

* make gotidy

* Mark GMP exporter beta

* Add changelog entries

* Update gcp exporter README

* Add tracking issue to chlog entries

* make crosslink

* make gotidy

* unit test fix

liustanley committed:
Update datadog exporter codeowners (#16554)

Add myself as a code owner

---------

Co-authored-by: Pablo Baeyens <pbaeyens31+github@gmail.com>

bogdandrutu committed:
[signalfxexporter] Deprecate config translation rules in favor of metrics transform processor (#18218)

Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

frzifus committed:
[extension/jaegerremotesampling] remove invalid comment (#18230)

Signed-off-by: Benedikt Bongartz <bongartz@klimlive.de>

bogdandrutu committed:
[k8sclusterreceiver] Split scrapers in separate packages, allows metadata per package (#18216)

Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

dmitryax committed:
[chore] [cmd/mdatagen] Don't use pmetric.Metrics.MoveTo (#18235)

The Metrics.MoveTo call is unnecessary. We can just pass the built Metrics buffer.

Commit:
[chore] Added missing details to CONTRIBUTING.md (#18213)

Explain how to update intra-repository dependencies when adding a new component.

TylerHelmuth committed:
[processor/transform] Remove unused structs (#18240)

Remove unused structs

khanhntd committed:
Instead of supply metric metadata and label for aws calculator, supply with aws metric key which consists of metric metadata and label (#17207)

* Change to use metric key instead of metric and labels

* Using GoImport to fix linter

* Using GoImport to fix linter

* Using GoImport to fix linter

* Change description and change name

* Change description and change name

yutingcaicyt committed:
[pkg/stanza/operator/transformer/recombine] Add the max_log_size config to recombine operator (#17387) (#18089)

"max_log_size" is added to the config of recombine operator. Once the total bytes size of the combined field exceeds the limit, all received entries of the source will be combined and flushed.

We've had discussion around this feature (see #17387). Here I choose the "soft limit" rather than truncating the log to get the fixed bytes size when total size exceeds the limit because I think soft limitation is enough for users on this occasion.

songy23 committed:
[exporter/datadog] Run API key validation in separate goroutine (#18239)

* [exporter/datadog] Run API key validation in separate goroutine

* Update exporter/datadogexporter/logs_exporter.go

Co-authored-by: Pablo Baeyens <pbaeyens31+github@gmail.com>

* Update exporter/datadogexporter/traces_exporter.go

Co-authored-by: Pablo Baeyens <pbaeyens31+github@gmail.com>

---------

Co-authored-by: Pablo Baeyens <pbaeyens31+github@gmail.com>

seankhliao committed:
[processor/cumulativetodelta]: always drop first metric point (#18224)

update tests to drop point 1

Commit:
[chore] make update-otel (#18186)

make update-otel

Signed-off-by: Alex Boten <aboten@lightstep.com>

Commit:
[chore] bump otel metrics dep (#18258)

Updating to 0.35.0 requires changes to accomodate the new API.

Signed-off-by: Alex Boten <aboten@lightstep.com>

nslaughter committed:
address issues with "flaky sftp test" (#18212)

Commit:
[chore] dependabot updates (#18260)

The dependabot action is failing currently, updating the remaining dependabot updates manually.

Signed-off-by: Alex Boten <aboten@lightstep.com>

boostchicken committed:
[receiver/datadog] DD Agent API Updates (#18228)

* new

* v0.7 and v0.5

* chlog

* tidy

* gofmt

* lint fixes

* Update go.mod

* Added unit test

* tudy

* tidy

* tidy

hvaghani221 committed:
[receiver/splunkhec] Drop events when nested indexed fields are present (#18068)

According to the Splunk HEC doc, fields key specifies a JSON object that contains a flat (not nested) list of explicit custom fields to be defined at index time. The current implementation allows receiving nested fields. This PR adds validation for nested fields. When nested fields are found, the entire batch is dropped and will return a 400 error code.

dmitryax committed:
[pkg/stanza/adapter] Remove deprecated Convert and ConvertFrom (#18259)

fatsheep9146 committed:
[pkg/pdatatest] update the content related to trace in doc (#18267)

Signed-off-by: Ziqi Zhao <zhaoziqi9146@gmail.com>

hvaghani221 committed:
[exporter/splunkhec] Flatten nested attribute map (#18241)

According to the Splunk HEC doc, the fields key specifies a JSON object that contains a flat (not nested) list of explicit custom fields to be defined at index time.
The current implementations send the nested map or array as it is. It leads to Error in handling indexed fields error.
In this PR, I have added to logic to flatten the nested map and drop nested arrays.

mizhexiaoxiao committed:
[exporter/loadbalancingexporter] update loadbalancing exporter readme (#18219)

* update readme of loadbalancing exporter

* Update exporter/loadbalancingexporter/README.md

Co-authored-by: Juraci Paixão Kröhling <juraci.github@kroehling.de>

---------

Co-authored-by: Juraci Paixão Kröhling <juraci.github@kroehling.de>

gbbr committed:
exporter/datadogexporter: allow more trace-agents (#18248)

* exporter/datadogexporter: allow more trace-agents

This change ensures that each receiver instantiated by the factory uses
its own pkg/trace/agent instance to facilitate multiple API endpoints.

* Add changelog

Frapschen committed:
[processor/servicegraph] deprecate unprefix label set (#18268)

* remove old label set

* add changelog

TylerHelmuth committed:
[pkg/ottl] Update OTTL docs (#18263)

* Update OTTL docs

* Update pkg/ottl/README.md

Co-authored-by: Evan Bradley <github@evanbradley.org>

* Update pkg/ottl/README.md

Co-authored-by: Evan Bradley <github@evanbradley.org>

* Update pkg/ottl/README.md

Co-authored-by: Evan Bradley <github@evanbradley.org>

* Update pkg/ottl/README.md

Co-authored-by: Evan Bradley <github@evanbradley.org>

---------

Co-authored-by: Evan Bradley <github@evanbradley.org>

sigilioso committed:
[chore] [processor/cumulativetodelta]: cleanup non-monotonic sums references (#18249)

* chore: extend tests and remove old references

* Extend processor tests to check that non-monotonic sums are not converted
* Remove misleading comment
* Remove an unneeded condition when detecting restarts

* test: fix test after rebasing

TylerHelmuth committed:
[pkg/ottl] Add ErrorMode configuration option (#18187)

* Add ErrorMode configuration option to OTTL

* Use Send by default

* Add changelog entry

* Save reordering for separate PR

* fix lint

* Update naming

* Move error handling to Statements

* Remove ErrorMode from the parser struct

* remove changelog entry

schmikei committed:
[receiver/mongodbatlas] Add Events API Support (#18237)

* Add events support for mongdobatlas receiver

bogdandrutu committed:
[signalfxexporter] avoid stringify every data point that does not match the filter (#18274)

Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

seankhliao committed:
[processor/cumulativetodelta] fix readme key for max_staleness (#18279)

seankhliao committed:
[processor/k8sattributesprocessor] fix doc formatting (#18277)

dmitryax committed:
[chore] [cmd/mdatagen] Remove unused struct tags (#18283)

The validation was moved to Unmarshal methods, so these tags any not being used anymore

djaglowski committed:
[connector/count] Add counting implementation  (#18104)

* [connector/count] Add basic implementation

Frapschen committed:
[exporter/clickhouse] Fix ut when clickhouse-go/v2 version bump up (#18284)

* version bump up

* remove redundant comment

* make gotidy

khyatigandhi6 committed:
Add Logicmonitor Exporter implementation (#16608)

Frapschen committed:
[exporter/clickhouse] Fix StartTime is not inserted for Gauge and Counter (#18288)

Fix StartTime is not inserted for Gauge and Counter

sincejune committed:
[chore] Fix typo in regex parser doc (#18291)

rogercoll committed:
[docs] fix receiver_creator supported pipelines (#18304)

djaglowski committed:
Fix dependabot entry for count connector (#18382)

fatsheep9146 committed:
[processor/k8sattributes] Add first version of k8s e2e tests (#17410)

Signed-off-by: Ziqi Zhao <zhaoziqi9146@gmail.com>
Co-authored-by: Dmitrii Anoshin <anoshindx@gmail.com>

dmitryax committed:
[chore] Rename e2e test and make them run on main branch and tags (#18387)

bogdandrutu committed:
Propose @atoulme as approver (#18309)

Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

bogdandrutu committed:
Promote @TylerHelmuth to maintainer (#18384)

Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

bogdandrutu committed:
Propose @kovrus as approver (#18308)

Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

jamesmoessis commented:
@mx-psi - that conversation hasn't budged in two weeks now, I don't mind if we keep the feature flag definition either way. Do you want me to do that and I can remove it later depending on the outcome of that conversation?

Commit:
[chore] update influx dep to 0.3.1 (#18396)

Fixes #18390

---------

Signed-off-by: Alex Boten <aboten@lightstep.com>

bogdandrutu committed:
Promote @fatsheep9146 to approver role (#18310)

Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>
Co-authored-by: Dmitrii Anoshin <anoshindx@gmail.com>

khyatigandhi6 committed:
Enable logicmonitor exporter (#18301)

* enable logicmonitor exporter

* Changing stability level to alpha

* fix failing checks: make genotelcontribcol

* reorder imports in go.mod

* reorder imports in go.mod

mx-psi commented:
> @mx-psi - that conversation hasn't budged in two weeks now. I'm happy to keep the feature flag definition to prevent errors. I can remove it later depending on the outcome of that conversation. Thoughts on that?

That makes sense to me, we can make this PR only remove the old implementation and merge it now, and keep the feature flag for later

mx-psi commented on receiver/dockerstatsreceiver/factory.go:
```diff
@@ -20,29 +20,17 @@ import (
+	stability = component.StabilityLevelAlpha
 )
 
-func init() {
```
so, to make my previous comment clear, I am fine with merging this PR if we don't remove this; everything else can be merged now IMO

songy23 committed:
[exporter/datadog] Enable retries when exporting usage metrics in Datadog traces exporter (#18282)

* [exporter/datadog] Enable retries in exportHostMetrics in Datadog traces_exporter

* Rename host metrics to usage metrics and update changelog

newly12 committed:
Refactor newMultilineSplitterFactory (#18303)

* refactor newMultilineSplitterFactory

boostchicken committed:
[receiver/datadog] Adopt SharedComponent (#18229)

boostchicken committed:
[receiver/datadog] Documentation Updates (#18398)

boostchicken committed:
[internal/components] add datadogreceiver to default build (#18399)

* Added Datadog to base build

* fixing to use automation

cartermp committed:
[receiver/datadog] Add telemetry.sdk.* resource attributes (#18265)

Commit:
[chore] remove realpath usage (#18410)

Trying to fix installation of tools on windows. It looks like SRC_ROOT isn't working as it is on windows:

```
/usr/bin/sh: line 1: /internal/tools/tools.go: No such file or directory
```

See https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/4113006957/jobs/7098614011

Signed-off-by: Alex Boten <aboten@lightstep.com>

schmikei committed:
[receiver/awscloudwatch] fix limit discovery exceeding API limits (#18306)

* Fix issue where limit per discovery request could be configured to exceed API limitations

* add blurb about how autodiscover.limit does not impact API batch size

opentelemetrybot committed:
[chore] dependabot updates Tue Feb  7 06:37:05 UTC 2023 (#18404)

Bump github.com/ClickHouse/clickhouse-go/v2 from 2.5.1 to 2.6.0 in /exporter/clickhouseexporter
Bump github.com/DataDog/agent-payload/v5 from 5.0.67 to 5.0.70 in /exporter/datadogexporter
Bump github.com/SAP/go-hdb from 0.111.9 to 1.0.2 in /receiver/saphanareceiver
Bump github.com/SAP/go-hdb from 0.111.9 to 1.0.2 in /receiver/sqlqueryreceiver
Bump github.com/antonmedv/expr from 1.10.5 to 1.12.0 in /pkg/stanza
Bump github.com/aws/aws-sdk-go from 1.44.192 to 1.44.194 in /exporter/awscloudwatchlogsexporter
Bump github.com/aws/aws-sdk-go from 1.44.192 to 1.44.194 in /exporter/datadogexporter
Bump github.com/aws/aws-sdk-go from 1.44.192 to 1.44.194 in /exporter/kafkaexporter
Bump github.com/aws/aws-sdk-go from 1.44.192 to 1.44.194 in /internal/aws/cwlogs
Bump github.com/aws/aws-sdk-go from 1.44.192 to 1.44.194 in /internal/aws/k8s
Bump github.com/aws/aws-sdk-go from 1.44.192 to 1.44.194 in /internal/aws/proxy
Bump github.com/aws/aws-sdk-go from 1.44.192 to 1.44.194 in /internal/metadataproviders
Bump github.com/aws/aws-sdk-go from 1.44.192 to 1.44.194 in /receiver/awscontainerinsightreceiver
Bump github.com/aws/aws-sdk-go-v2 from 1.17.3 to 1.17.4 in /exporter/awskinesisexporter
Bump github.com/aws/aws-sdk-go-v2 from 1.17.3 to 1.17.4 in /extension/sigv4authextension
Bump github.com/aws/aws-sdk-go-v2/config from 1.18.10 to 1.18.12 in /exporter/awskinesisexporter
Bump github.com/aws/aws-sdk-go-v2/config from 1.18.10 to 1.18.12 in /extension/sigv4authextension
Bump github.com/aws/aws-sdk-go-v2/credentials from 1.13.10 to 1.13.12 in /exporter/awskinesisexporter
Bump github.com/aws/aws-sdk-go-v2/credentials from 1.13.10 to 1.13.12 in /extension/sigv4authextension
Bump github.com/aws/aws-sdk-go-v2/service/kinesis from 1.17.2 to 1.17.3 in /exporter/awskinesisexporter
Bump github.com/aws/aws-sdk-go-v2/service/sts from 1.18.2 to 1.18.3 in /extension/sigv4authextension
Bump github.com/docker/docker from 20.10.23+incompatible to 23.0.0+incompatible in /internal/docker
Bump github.com/docker/docker from 20.10.23+incompatible to 23.0.0+incompatible in /internal/metadataproviders
Bump github.com/shirou/gopsutil/v3 from 3.22.12 to 3.23.1 in /receiver/awscontainerinsightreceiver
Bump github.com/shirou/gopsutil/v3 from 3.22.12 to 3.23.1 in /receiver/hostmetricsreceiver
Bump github.com/shirou/gopsutil/v3 from 3.22.12 to 3.23.1 in /testbed
Bump github.com/sijms/go-ora/v2 from 2.5.25 to 2.5.27 in /receiver/sqlqueryreceiver
Bump github.com/tencentcloud/tencentcloud-sdk-go/tencentcloud/common from 1.0.583 to 1.0.588 in /exporter/tencentcloudlogserviceexporter
Bump github.com/vmware/govmomi from 0.30.1 to 0.30.2 in /receiver/vcenterreceiver
Bump go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracehttp from 1.11.2 to 1.12.0 in /cmd/telemetrygen

---------

Signed-off-by: Alex Boten <aboten@lightstep.com>

Commit:
[chore] dependabot pr needs to run make genotel* (#18439)


Signed-off-by: Alex Boten <aboten@lightstep.com>
Co-authored-by: Bogdan Drutu <lazy@splunk.com>

opentelemetrybot committed:
[chore] dependabot updates Tue Feb  7 21:37:25 UTC 2023 (#18437)

* dependabot updates Tue Feb  7 21:37:22 UTC 2023
Bump docker/build-push-action from 3 to 4
Bump github.com/aws/aws-sdk-go from 1.44.194 to 1.44.196 in /exporter/awscloudwatchlogsexporter
Bump github.com/aws/aws-sdk-go from 1.44.194 to 1.44.196 in /exporter/awsemfexporter
Bump github.com/aws/aws-sdk-go from 1.44.194 to 1.44.196 in /exporter/awsxrayexporter
Bump github.com/aws/aws-sdk-go from 1.44.194 to 1.44.196 in /exporter/datadogexporter
Bump github.com/aws/aws-sdk-go from 1.44.194 to 1.44.196 in /exporter/kafkaexporter
Bump github.com/aws/aws-sdk-go from 1.44.194 to 1.44.196 in /extension/observer/ecsobserver
Bump github.com/aws/aws-sdk-go from 1.44.194 to 1.44.196 in /internal/aws/awsutil
Bump github.com/aws/aws-sdk-go from 1.44.194 to 1.44.196 in /internal/aws/cwlogs
Bump github.com/aws/aws-sdk-go from 1.44.194 to 1.44.196 in /internal/aws/k8s
Bump github.com/aws/aws-sdk-go from 1.44.194 to 1.44.196 in /internal/aws/proxy
Bump github.com/aws/aws-sdk-go from 1.44.194 to 1.44.196 in /internal/aws/xray
Bump github.com/aws/aws-sdk-go from 1.44.194 to 1.44.196 in /internal/aws/xray/testdata/sampleapp
Bump github.com/aws/aws-sdk-go from 1.44.194 to 1.44.196 in /internal/metadataproviders
Bump github.com/aws/aws-sdk-go from 1.44.194 to 1.44.196 in /processor/resourcedetectionprocessor
Bump github.com/aws/aws-sdk-go from 1.44.194 to 1.44.196 in /receiver/awscontainerinsightreceiver
Bump github.com/aws/aws-sdk-go from 1.44.194 to 1.44.196 in /receiver/awsxrayreceiver
Bump github.com/golangci/golangci-lint from 1.50.1 to 1.51.1 in /internal/tools
Bump github.com/tencentcloud/tencentcloud-sdk-go/tencentcloud/common from 1.0.588 to 1.0.589 in /exporter/tencentcloudlogserviceexporter
Bump go.etcd.io/bbolt from 1.3.6 to 1.3.7 in /extension/storage
Bump go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracehttp from 1.12.0 to 1.13.0 in /cmd/telemetrygen

* make gen (#18438)

* remove unused var, bump build-push-action (#18440)

- remove unused var
- update build-push-action to 4

Signed-off-by: Alex Boten <aboten@lightstep.com>

* linting issue (#18441)

Signed-off-by: Alex Boten <aboten@lightstep.com>

* linting issue (#18442)

Signed-off-by: Alex Boten <aboten@lightstep.com>

---------

Signed-off-by: Alex Boten <aboten@lightstep.com>
Co-authored-by: Alex Boten <aboten@lightstep.com>

Commit:
[chore] make update-otel  (#18443)


---------

Signed-off-by: Alex Boten <aboten@lightstep.com>

jamesmoessis committed:
Merge branch 'remove-dockerstats-featuregate' of github.com:atlassian-forks/opentelemetry-collector-contrib into remove-dockerstats-featuregate

jamesmoessis committed:
reconciliate changes with removal of init()

jamesmoessis commented:
Not sure what happened with the commit history there. I will close this and re open.
